stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/security/KeyStore.java

```diff
@@ -0,0 +1,256 @@
+                .toCharArray();
+        String keyAlias = configRegistry.getInstanceConfigValue(KEY_STORE_CONFIG, KEY_ALIAS);
+        return getPrivateKey(keyAlias, keyStorePassword);
+    }
+
+    /**
+     * Get the default public key for the service.
+     *
+     * @return default public key
+     * @throws KeyStoreException if the keystore has not been initialized
+     *                           (loaded), or this operation fails for some other reason
+     */
+    public PublicKey getDefaultPublicKey() throws KeyStoreException {
+        String keyAlias = ConfigRegistry.getInstance().getInstanceConfigValue(KEY_STORE_CONFIG, KEY_ALIAS);
+        return getPublicKey(keyAlias);
+    }
+
+    /**
+     * Get the default certificate for the service.
+     *
+     * @return certificate corresponding to the alias
+     * @throws KeyStoreException if the keystore has not been initialized
+     *                           (loaded), or this operation fails for some other reason
+     */
+    public Certificate getDefaultCertificate() throws KeyStoreException {
+        String keyAlias = ConfigRegistry.getInstance().getInstanceConfigValue(KEY_STORE_CONFIG, KEY_ALIAS);
+        return getCertificate(keyAlias);
+    }
+
+    private void loadKeyStore() {
+        ConfigRegistry configRegistry = ConfigRegistry.getInstance();
+        String keyStoreLocation = configRegistry.getInstanceConfigValue(KEY_STORE_CONFIG, KEY_STORE_LOCATION);
+        if (keyStoreLocation != null) {
+            char[] keyStorePassword = configRegistry.getInstanceConfigValue(
+                    KEY_STORE_CONFIG, KEY_STORE_PASSWORD).toCharArray();
+            String keystoreType = configRegistry.getInstanceConfigValue(KEY_STORE_CONFIG, KEY_STORE_TYPE);
+            try (InputStream file = new FileInputStream(new File(keyStoreLocation))) {
+                keyStore = java.security.KeyStore.getInstance(keystoreType);
+                keyStore.load(file, keyStorePassword);
+            } catch (FileNotFoundException e) {
+                throw new BallerinaException("Failed to load keystore: file not found: " + keyStoreLocation, e);
+            } catch (Exception e) {
+                throw new BallerinaException("Failed to load keystore: " + e.getMessage(), e);
+            }
+        } else {
+            try {
+                log.warn("KeyStore is not configured");
+                keyStore = java.security.KeyStore.getInstance("pkcs12");
+            } catch (java.security.KeyStoreException e) {
+                throw new BallerinaException("Failed to instantiate keystore: ", e);
+            }
+        }
+    }
+
+    private void loadTrustStore() {
+        ConfigRegistry configRegistry = ConfigRegistry.getInstance();
+        String trustStoreLocation = configRegistry.getInstanceConfigValue(TRUST_STORE_CONFIG, TRUST_STORE_LOCATION);
+        if (trustStoreLocation != null) {
```

Comment:
Can this be empty?

Reply:
This should not be null. This is used until registry API provides a method to check whether main config block exists.